processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -189,6 +189,11 @@ public boolean isVoid() {
         return isVoid;
     }
 
+    public boolean isAbstract() {
```

Comment:
Would that actually kick in for interfaces, too? I don't think we need keep the two cases apart, interfaces are just a form of abstract type. In the error message we could say "result type may not be an abstract class nor interface".